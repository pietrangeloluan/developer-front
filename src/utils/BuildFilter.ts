export default function (params: any) {
  let filter = {}
  for (const key in params) {
    if (Object.hasOwnProperty.call(params, key)) {
      const parameter = params[key]
      if (parameter) {
        filter = { ...filter, [key]: parameter }
      }
    }
  }
  return filter
}
