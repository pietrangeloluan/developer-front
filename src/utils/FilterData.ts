export default function (obj, filter) {
  const filtered = obj.filter(item => {
    for (const key in filter) {
      if (item[key] === undefined || !filter[key].includes(item[key])) {
        return false
      }
    }
    return true
  })

  return filtered
}
