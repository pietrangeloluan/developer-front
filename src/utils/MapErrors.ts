export function mapErrors(errors, context) {
  const errorKeys = Object.keys(errors)
  const regex = new RegExp(`${context}\\[\\d+\\]`, 'g')
  const contextErrors = errorKeys.filter(err => err.match(regex))

  if (!contextErrors.length) return [{}]

  const updated = []

  contextErrors.forEach(err => {
    const found = [...err.matchAll(/\[(\d+)\]\.(\w+)/g)]

    if (found.length) {
      found[0].shift()
      const errIndex = found[0][0]
      const errField = found[0][1]

      updated[errIndex] = {
        ...updated[errIndex],
        [errField]: errors[`${context}[${errIndex}].${errField}`]
      }
    }
  })

  return updated
}
