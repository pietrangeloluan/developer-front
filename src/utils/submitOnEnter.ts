import { KeyboardEvent } from 'react'

export function submitOnEnter(fn: Function, args: any[] = []) {
  return function (e: KeyboardEvent) {
    if (e.key === 'Enter') fn.apply(args)
  }
}
