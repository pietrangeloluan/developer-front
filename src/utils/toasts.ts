import { toast } from 'react-toastify'

export const toasts = {
  created() {
    return toast.success('Dados cadastrados com sucesso.', {
      autoClose: 5 * 1000
    })
  },
  deleted() {
    return toast.success('Dados exclu√≠dos com sucesso.', {
      autoClose: 5 * 1000
    })
  },
  edited() {
    return toast.success('Dados editados com sucesso.', { autoClose: 5 * 1000 })
  },
  generalFail({ code = 'UNKNOWN_ERROR' } = {}) {
    return toast.error(
      'Ocorreu uma falha inesperada. Por favor, aguarde alguns minutos de tente novamente.',
      { autoClose: 5 * 1000, toastId: code }
    )
  }
}
