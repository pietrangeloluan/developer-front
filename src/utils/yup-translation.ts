/* eslint-disable no-template-curly-in-string */
import { setLocale } from 'yup'

const translation = {
  array: {
    max: '${path} deve conter no máximo ${max} itens',
    min: '${path} deve conter no mínimo ${min} itens'
  },
  date: {
    max: '${path} deve ser anterior à ${max}',
    min: '${path} deve ser mais recente que ${min}',
    required: '${path} é obrigatório'
  },
  mixed: {
    default: '${path} é inválido',
    notOneOf: '${path} não pode ser igual à um destes valores: ${values}',
    oneOf: '${path} deve ser igual à um dos seguintes valores: ${values}',
    required: '${path} é obrigatório'
  },
  number: {
    integer: '${path} deve ser inteiro',
    lessThan: '${path} deve ser menor à ${less}',
    max: '${path} deve ser no máximo ${max}',
    min: '${path} deve ser no mínimo ${min}',
    moreThan: '${path} deve ser maior à ${more}',
    negative: '${path} deve ser negativo',
    notEqual: '${path} não pode ser igual à ${notEqual}',
    positive: '${path} deve ser positivo',
    required: '${path} é obrigatório'
  },
  string: {
    email: '${path} não contém um e-mail válido',
    length: '${path} deve conter exatamente ${length} caracteres',
    lowercase: '${path} deve ser maiúsculo',
    max: '${path} deve conter no máximo ${max} caracteres',
    min: '${path} deve conter pelo menos ${min} caracteres',
    trim: '${path} não deve conter espaços em branco em início ou final.',
    uppercase: '${path} deve ser minúsculo',
    url: '${path} não contém uma URL válida'
  }
}

setLocale(translation)
