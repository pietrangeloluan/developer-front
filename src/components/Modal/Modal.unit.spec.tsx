import { fireEvent, screen } from '@testing-library/react'

import { renderWithThemeProvider } from '__tests__/utils'

import { Modal } from '.'

describe('Modal', () => {
  it('should render the component', () => {
    renderWithThemeProvider(Modal)

    const found = screen.getByTestId('modal')
    expect(found).toBeInTheDocument()
  })

  it('should match the snapshot', () => {
    const { container } = renderWithThemeProvider(Modal)
    expect(container).toMatchSnapshot()
  })

  it('should has no action button', () => {
    renderWithThemeProvider(Modal, {
      isActionsEnabled: false
    })
    const modal = screen.getByTestId('modal')
    const actions = screen.getByTestId('no-actions')

    expect(modal).toContainElement(actions)
  })

  it('should handle open modal ', () => {
    renderWithThemeProvider(Modal)
    const modal = screen.getByTestId('modal')

    expect(modal.getAttribute('aria-hidden')).toBe('false')
    expect(modal).toHaveStyle({ opacity: 1 })
  })

  it('should handle close modal when cancel button clicked', () => {
    renderWithThemeProvider(Modal)
    const modal = screen.getByTestId('modal')

    fireEvent.click(screen.getByTestId('close-modal'))

    expect(modal.getAttribute('aria-hidden')).toBe('true')
    expect(modal).toHaveStyle({ opacity: 0 })
  })
})
