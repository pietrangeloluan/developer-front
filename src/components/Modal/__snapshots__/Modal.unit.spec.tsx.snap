// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match the snapshot 1`] = `
.c0 {
  z-index: 202 !important;
  width: 100%;
  height: 100%;
  background: #00000047;
  position: fixed;
  top: 0;
  left: 0;
  opacity: 1;
}

.c1 {
  max-width: 60rem;
  margin: 0 auto;
}

.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  vertical-align: center;
  background: white;
  border: none;
  border-radius: 2rem;
  box-shadow: 0px 30px 63px #00000052;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  -webkit-transition: opacity '0.3s ease-in-out';
  transition: opacity '0.3s ease-in-out';
  width: 60rem;
  z-index: 201;
  bottom: 50%;
  -webkit-transform: translate(0,50%);
  -ms-transform: translate(0,50%);
  transform: translate(0,50%);
  opacity: 1;
}

.c2 .title {
  color: #3684ab;
  font-size: 1.8rem;
  font-weight: 600;
}

.c3 {
  color: #3684ab;
  position: absolute;
  right: 0;
  top: 0;
  padding: 2.4rem;
  cursor: pointer;
  text-align: end;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c6 {
  width: 60rem;
  max-height: 80rem;
  padding: 2.4rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 3.2rem 1.6rem 0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding-top: 1.6rem;
  padding-bottom: 1.6rem;
  -webkit-align-self: flex-end;
  -ms-flex-item-align: end;
  align-self: flex-end;
}

.c9 button {
  margin-left: 1.6rem;
}

.c8 {
  margin: 0 0.8rem;
  width: auto;
  height: 0.2rem;
  background-color: #f2f3f7;
}

.c4 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 16px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 16px;
}

.c10 {
  outline: none;
  border-radius: 0.5rem;
  border: none;
  font-size: 1.4rem;
  font-weight: 600;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
  -webkit-transition: background-color 400ms,color 400ms,-webkit-transform 400ms;
  -webkit-transition: background-color 400ms,color 400ms,transform 400ms;
  transition: background-color 400ms,color 400ms,transform 400ms;
  will-change: background-color,color,transform;
  margin: 0.8rem;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
}

.c10.blue {
  background-color: #3684ab;
  color: white;
}

.c10.blue:hover {
  background-color: #00577c;
}

.c10.green {
  background-color: #3fab36;
  color: white;
}

.c10.green:hover {
  background-color: #007b00;
}

.c10.gray {
  background-color: #707070;
  color: white;
}

.c10.gray:hover {
  background-color: #454545;
}

.c10:active {
  -webkit-transform: scale(0.88);
  -ms-transform: scale(0.88);
  transform: scale(0.88);
}

.c10.large {
  padding: 1.4rem 10rem;
  font-size: 1.6rem;
}

.c10.medium {
  padding: 1.1rem 3.2rem;
}

.c10.small {
  padding: 0.7rem 2.8rem;
}

.c10.xsmall {
  box-shadow: 0px 3px 6px #c1c5c4;
  font-weight: 400;
  padding: 0.1rem 1.1rem;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
      width="60"
    >
      <div
        aria-hidden="false"
        aria-label="modal"
        class="c2"
        data-testid="modal"
        width="60"
      >
        <div
          aria-label="close-modal"
          class="c3"
          data-testid="close-modal"
        >
          <div
            class="c4"
            height="16"
            preserveaspectratio="xMidYMid meet"
            width="16"
          >
            <div
              data-src="/icons/modal-exit.svg"
              src=""
            />
          </div>
        </div>
        <div
          class="c5"
        >
          <div
            class="c6"
            width="60"
          />
          <div
            class="c7 actions"
          >
            <div
              class="c8"
            />
            <div
              class="c9"
            >
              <div>
                <button
                  class="c10 gray medium"
                >
                  <p>
                    Cancelar
                  </p>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
